std/special/compiler_rt/floatuntisf.zig

```diff
@@ -0,0 +1,59 @@
+const builtin = @import("builtin");
+const is_test = builtin.is_test;
+
+const FLT_MANT_DIG = 24;
+
+pub extern fn __floatuntisf(arg: u128) f32 {
+    @setRuntimeSafety(is_test);
+
+    if (arg == 0)
+        return 0.0;
+
+    var a = arg;
+    const N: u32 = @sizeOf(u128) * 8;
+    const sd = @bitCast(i32, N -% @clz(a)); // number of significant digits
+    var e: i32 = sd -% 1; // exponent
+    if (sd > FLT_MANT_DIG) {
+        //  start:  0000000000000000000001xxxxxxxxxxxxxxxxxxxxxxPQxxxxxxxxxxxxxxxxxx
+        //  finish: 000000000000000000000000000000000000001xxxxxxxxxxxxxxxxxxxxxxPQR
+        //                                                12345678901234567890123456
+        //  1 = msb 1 bit
+        //  P = bit FLT_MANT_DIG-1 bits to the right of 1
+        //  Q = bit FLT_MANT_DIG bits to the right of 1
+        //  R = "or" of all bits to the right of Q
+        switch (sd) {
+            FLT_MANT_DIG + 1 => {
+                a <<= 1;
+            },
+            FLT_MANT_DIG + 2 => {},
+            else => {
+                const shift_amt = @bitCast(i32, N +% (FLT_MANT_DIG + 2)) -% sd;
```

Comment:
For clarity: `N + FLT_MANT_DIG + 2)`?  When I see `%+`, I immediately go scan for possible overflow inputs but there can't be any in this case, can it?  I'd use wraparound operators sparingly.

Line 14: same comment.  Also maybe line 32, 38, 43, etc.

Reply:
I agree. For the moment I used existing compiler_rt functions as a template and have done minimal changes to the actual structure. I'll do a more in-depth scan when complete and make these sorts of things more idiomatic/correct.